File: frontend/src/types/warehouse.ts
import type { PageQuery } from './common'

/** 仓库档案，与后端 Warehouse 对应（含 BaseEntity 通用字段，按需取用）。 */
export interface Warehouse {
  uuid: string
  warehouseCode?: string
  warehouseName: string
  /** 库位/地址 */
  location?: string
  /** 1启用 2停用 */
  status?: number
  isDefault?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

/** 仓库新增/修改入参，与后端 WarehouseSaveDTO 对应。 */
export interface WarehouseSaveDTO {
  warehouseCode?: string
  warehouseName: string
  location?: string
  status?: number
  isDefault?: number
  remark?: string
}

/** 仓库列表查询入参。 */
export interface WarehouseQuery extends PageQuery {
  keyword?: string
  status?: number
}

File: src/main/java/com/paper/mes/warehouse/entity/Warehouse.java
package com.paper.mes.warehouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.paper.mes.common.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 仓库档案 sys_warehouse。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_warehouse")
public class Warehouse extends BaseEntity {

    @TableId(type = IdType.ASSIGN_UUID)
    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    private String uuid;

    private String warehouseCode;
    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
    private String warehouseName;
    /** 仓库地址或识别说明，不表示库位层级。 */
    private String location;
    /** 1启用 2停用 */
    private Integer status;
    private Integer isDefault;
    private String remark;
}
